### DISCUSSION TITLE
(Request) Can we get glass versions of every hull block

### DISCUSSION THREAD

**Discussion by Arutus** [tsunamayo/Starship-EVO#4094] [Suggestions]
as the current glass doesn't not line up and has bad collisions and I feel a glass version of every hull block would make for much better bridge work what with the current scaling.

**Comment by tsunamayo** [comment]
@Arutus what shape do you want? Could you show me example of what you want to build, to me basic shape use seems rather limited for transparent bits, but I could always add them. Thanks

**Reply by ProPeach** [reply]
The most important shape I think we're missing is a canopy version of the hepta shape.
![image](https://user-images.githubusercontent.com/27857711/128177764-fe837930-264e-40a7-8fbf-311135fc3ecc.png)

This is pretty much integral to most canopy shapes that aren't just a very simple wedge. Most builders use an inverted tetra brick instead, which makes a large bump on the side of your canopy
![image](https://user-images.githubusercontent.com/27857711/128177847-00a854ed-09c9-4311-96d7-de43012a838d.png)
![image](https://user-images.githubusercontent.com/27857711/128178551-eb7933e0-01ed-4bcd-b451-cf7030284ee9.png)


In regards to the collisions, the tetra shape is the one that causes the most issues from what I've seen. Here, you can't fill in this gap with a canopy wedge because of it
![image](https://user-images.githubusercontent.com/27857711/128178075-96f4363c-babc-449a-93fa-5785695cdb6c.png)

Nor can you build any consoles/shapes inside a tetra either
![image](https://user-images.githubusercontent.com/27857711/128178135-98fed524-fc93-47a9-a829-fb8544779c6b.png)


Blueprint here - [Canopy Examples.zip](https://github.com/tsunamayo/Starship-EVO/files/6931063/Canopy.Examples.zip)

As an aside, do you think there's much you can do about the borders that you see through the glass? I completely understand why they're there, and I remember that you mentioned how difficult "connected textures" like that on bricks would be, but they do get in the way of an otherwise fantastic set of tools
![image](https://user-images.githubusercontent.com/27857711/128178370-9366971e-33ed-4e97-96ae-09ed7d546a60.png)

**Comment by PFSCommissar** [comment]
this may not be practical, but is there any way we can remove the seams when two glass blocks are connected, so the windows look more like one solid piece?

**Comment by BruceLeedleLeedleLeedleLee** [comment]
I agree with the others a hepta and the seams would greatly improve cockpits, without all the annoying flickering

Id also like to suggest a form of """"penta"""" but hollowed 
![20210804144314_1](https://user-images.githubusercontent.com/37705129/128182879-d069a5e5-c234-4d72-83c8-0c423a5663a8.jpg)
![20210804144309_1](https://user-images.githubusercontent.com/37705129/128182894-2c7865ee-97a7-4715-b3ec-4df784833552.jpg)
Similar to the current corner it would probably have to be twice the size of the grid size but it will allow for nice chunky cockpits or car windshields

**Reply by tsunamayo** [reply]
@BruceLeedleLeedleLeedleLee so the flickering is due to random draw order, hopefully one day I improve the rendering tech to deal with this.

**Comment by Calvert4096** [comment]
I second this request... I was exploiting lighting to make a translucent-looking bussard collector effect.  This ceased working following a recent lighting update that (reasonably) prevents lights from bleeding through solid blocks.  I'm planning to replace the dome with glass, but as other have pointed out, the limited variety of glass blocks combined with the visible edges make this not look very good.

https://i.imgur.com/IkPjaHT.mp4

The closer I can get to the OG bussard collector look, the better.  The existing frosted glass material would be fantastic for this, if I could make a seamless dome out of that material.

https://youtu.be/wa79FmRRgJ0?t=69
